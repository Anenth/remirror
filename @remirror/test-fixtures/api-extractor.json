{
  "__AUTO_GENERATED__": "To update run: `pnpm generate:json`",
  "extends": "../../support/api-extractor.json",
  "mainEntryPointFilePath": "<projectFolder>/lib/index.d.ts",
  "apiReport": {
    "enabled": true,
    "reportFolder": "../../support/api/",
    "reportFileName": "remirror__test-fixtures.api.md"
  }
}
