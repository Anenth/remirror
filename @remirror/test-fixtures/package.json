{
  "name": "@remirror/test-fixtures",
  "version": "0.0.4",
  "private": true,
  "description": "Test fixtures for remirror unit tests",
  "sideEffects": false,
  "main": "dist/test-fixtures.cjs.js",
  "module": "dist/test-fixtures.esm.js",
  "types": "dist/test-fixtures.cjs.d.ts",
  "files": [
    "dist"
  ],
  "dependencies": {
    "@babel/runtime": "^7",
    "@remirror/core": "^0.11.0",
    "@remirror/extension-doc": "^0.11.0",
    "@remirror/extension-paragraph": "^0.11.0",
    "@remirror/extension-text": "^0.11.0",
    "@remirror/pm": "^0.0.0",
    "@remirror/preset-core": "^0.11.0",
    "@remirror/ui": "^0.7.6",
    "min-document": "^2.19.0"
  }
}
