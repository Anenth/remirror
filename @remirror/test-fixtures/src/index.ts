export * from './object-nodes';
export * from './object-shapes';
export * from './typecheck';
export * from './editor';
export * from './library';
