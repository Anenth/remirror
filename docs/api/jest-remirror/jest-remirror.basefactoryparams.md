<!-- Do not edit this file. It is automatically generated by API Documenter. -->

[Home](./index.md) &gt; [jest-remirror](./jest-remirror.md) &gt; [BaseFactoryParams](./jest-remirror.basefactoryparams.md)

## BaseFactoryParams interface

<b>Signature:</b>

```typescript
export interface BaseFactoryParams<GSchema extends EditorSchema = EditorSchema> extends Partial<AttrsParams> 
```

## Properties

|  Property | Type | Description |
|  --- | --- | --- |
|  [name](./jest-remirror.basefactoryparams.name.md) | <code>string</code> | The name of the node or mark |
|  [schema](./jest-remirror.basefactoryparams.schema.md) | <code>GSchema</code> | The editor schema |

