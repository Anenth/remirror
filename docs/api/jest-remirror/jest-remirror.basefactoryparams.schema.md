<!-- Do not edit this file. It is automatically generated by API Documenter. -->

[Home](./index.md) &gt; [jest-remirror](./jest-remirror.md) &gt; [BaseFactoryParams](./jest-remirror.basefactoryparams.md) &gt; [schema](./jest-remirror.basefactoryparams.schema.md)

## BaseFactoryParams.schema property

The editor schema

<b>Signature:</b>

```typescript
schema: GSchema;
```
