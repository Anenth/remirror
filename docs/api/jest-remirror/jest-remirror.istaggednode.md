<!-- Do not edit this file. It is automatically generated by API Documenter. -->

[Home](./index.md) &gt; [jest-remirror](./jest-remirror.md) &gt; [isTaggedNode](./jest-remirror.istaggednode.md)

## isTaggedNode variable

Check if a node is tagged.

<b>Signature:</b>

```typescript
isTaggedNode: (val: unknown) => val is TaggedProsemirrorNode<import("@remirror/core").EditorSchema<string, string>>
```
