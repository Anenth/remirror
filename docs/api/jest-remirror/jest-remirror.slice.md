<!-- Do not edit this file. It is automatically generated by API Documenter. -->

[Home](./index.md) &gt; [jest-remirror](./jest-remirror.md) &gt; [slice](./jest-remirror.slice.md)

## slice variable

<b>Signature:</b>

```typescript
slice: (schema: EditorSchema<string, string>) => (...content: TaggedContentWithText[]) => Slice<EditorSchema<string, string>>
```
