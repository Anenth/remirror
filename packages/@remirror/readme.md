# @remirror

This folder contains all the scoped packages under the `@remirror/` npm organisation.
