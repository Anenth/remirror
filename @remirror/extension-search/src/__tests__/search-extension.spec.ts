import { renderEditor } from 'jest-remirror';

import { isExtensionValid } from '@remirror/test-fixtures';

import { SearchExtension, SearchOptions } from '..';

test('is valid', () => {
  expect(isExtensionValid(SearchExtension, {}));
});

function create(options?: SearchOptions) {
  const {
    add,
    nodes: { p, doc },
    view,
    commands,
  } = renderEditor([new SearchExtension(options)]);
  const node = doc(p('welcome'), p('friend'), p('hello'), p('welcome friend'));

  return { add, p, doc, view, commands, node };
}

describe('commands', () => {
  test('#find', () => {
    const { view, add, commands, node } = create();
    add(node);

    commands.find('welcome');
    expect(view.dom).toMatchSnapshot();

    commands.findNext();
    expect(view.dom).toMatchSnapshot();

    commands.findPrevious();
    expect(view.dom).toMatchSnapshot();
  });

  test('#replace', () => {
    const { view, add, commands, node } = create();
    add(node);

    commands.find('we');
    commands.replace('me');
    expect(view.dom).toMatchSnapshot();
  });

  test('#replaceAll', () => {
    const { view, add, commands, node } = create();
    add(node);

    commands.find('friend');
    commands.replaceAll('enemy');
    expect(view.dom).toMatchSnapshot();
  });

  test('#clearSearch', () => {
    const { view, add, commands, node } = create();
    add(node);

    commands.find('friend');
    commands.clearSearch();
    expect(view.dom).toMatchSnapshot();
  });
});
