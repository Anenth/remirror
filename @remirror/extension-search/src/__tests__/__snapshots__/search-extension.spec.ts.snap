// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`commands #clearSearch 1`] = `
<div
  aria-label=""
  aria-multiline="true"
  autofocus="false"
  class="ProseMirror remirror-editor"
  contenteditable="true"
  role="textbox"
>
  <p>
    welcome
  </p>
  <p>
    friend
  </p>
  <p>
    hello
  </p>
  <p>
    welcome friend
  </p>
</div>
`;

exports[`commands #find 1`] = `
<div
  aria-label=""
  aria-multiline="true"
  autofocus="false"
  class="ProseMirror remirror-editor"
  contenteditable="true"
  role="textbox"
>
  <p>
    <span
      class="search highlighted-search"
    >
      welcome
    </span>
  </p>
  <p>
    friend
  </p>
  <p>
    hello
  </p>
  <p>
    <span
      class="search"
    >
      welcome
    </span>
     friend
  </p>
</div>
`;

exports[`commands #find 2`] = `
<div
  aria-label=""
  aria-multiline="true"
  autofocus="false"
  class="ProseMirror remirror-editor"
  contenteditable="true"
  role="textbox"
>
  <p>
    <span
      class="search"
    >
      welcome
    </span>
  </p>
  <p>
    friend
  </p>
  <p>
    hello
  </p>
  <p>
    <span
      class="search highlighted-search"
    >
      welcome
    </span>
     friend
  </p>
</div>
`;

exports[`commands #find 3`] = `
<div
  aria-label=""
  aria-multiline="true"
  autofocus="false"
  class="ProseMirror remirror-editor"
  contenteditable="true"
  role="textbox"
>
  <p>
    <span
      class="search highlighted-search"
    >
      welcome
    </span>
  </p>
  <p>
    friend
  </p>
  <p>
    hello
  </p>
  <p>
    <span
      class="search"
    >
      welcome
    </span>
     friend
  </p>
</div>
`;

exports[`commands #replace 1`] = `
<div
  aria-label=""
  aria-multiline="true"
  autofocus="false"
  class="ProseMirror remirror-editor"
  contenteditable="true"
  role="textbox"
>
  <p>
    <span
      class="search highlighted-search"
    >
      me
    </span>
    lcome
  </p>
  <p>
    friend
  </p>
  <p>
    hello
  </p>
  <p>
    <span
      class="search"
    >
      we
    </span>
    lcome friend
  </p>
</div>
`;

exports[`commands #replaceAll 1`] = `
<div
  aria-label=""
  aria-multiline="true"
  autofocus="false"
  class="ProseMirror remirror-editor"
  contenteditable="true"
  role="textbox"
>
  <p>
    welcome
  </p>
  <p>
    <span
      class="search highlighted-search"
    >
      enemy
    </span>
  </p>
  <p>
    hello
  </p>
  <p>
    welcome 
    <span
      class="search"
    >
      enemy
    </span>
  </p>
</div>
`;
